src/System.Management.Automation/engine/SessionStateContainer.cs

```diff
@@ -1853,8 +1853,16 @@ internal Collection<PSObject> GetChildItems(string[] paths, bool recurse, uint d
 
             // If the item is a container we have to filter its children
             // Use a hint + lazy evaluation to skip a container check
-
-            if (skipIsItemContainerCheck || IsItemContainer(providerInstance, path, context))
+            bool itemContainer = false;
+            try
+            {
```

Comment:
@nitinbps would it make sense to wrap this call into a function to avoid code duplication?